src/accountsdb/buffer_pool.zig

```diff
@@ -604,32 +709,32 @@ pub const HierarchicalFIFO = struct {
     ) error{InvalidKey}!void {
         if (key == INVALID_FRAME) return error.InvalidKey;
 
-        switch (metadata.in_queue[key]) {
+        switch (metadata.in_queue[key].load(.acquire)) {
             .main, .small => {
-                metadata.freq[key] +|= 1;
+                metadata.freqIncrement(key);
             },
             .ghost => {
-                std.debug.assert(metadata.freq[key] == 0);
-                metadata.freq[key] = 1;
+                std.debug.assert(metadata.freqIsZero(key));
+                metadata.freqSetToOne(key);
                 // Add key to main too - important to note that the key *still*
                 // exists within ghost, but from now on we'll ignore that entry.
                 self.main.writeItemAssumeCapacity(key);
-                metadata.in_queue[key] = .main;
+                metadata.in_queue[key].store(.main, .release);
```

Comment:
If this is callable by multiple threads, there could be spurious `writeItemAssumeCapacity` calls. Or the `freqIsZero` could work on one thread but fail on another.

Might be better to wrap the whole function inside a mutex since theres multiple changes that need to be atomic w.r.t. each other.

Reply:
now behind a mutex https://github.com/Syndica/sig/commit/e91f7ca39c43bf182de25213b9136a8a9d854ef3